clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -46,23 +47,28 @@ public HubConnection(String url, Transport transport, Logger logger, boolean ski
         }
         this.skipNegotiate = skipNegotiate;
         this.callback = (payload) -> {
+            hubConnectionStateLock.lock();
```

Comment:
Why is the lock being used here?

Reply:
We used to change the HubConnectionState here. Good find